File: 3st/Hyeonseo/programmer/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public List<?> solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        answer.add(arr[0]);
        for(int i = 1; i < arr.length; i++) {
            if(arr[i-1] != arr[i]) {
                answer.add(arr[i]);
            }
        }     
        return answer;
    }
}
//level 1

File: 4st/Seohyun/같은 숫자는 싫어.js
function solution(arr) {
  let answer = [];

  for (let n of arr) {
    if (answer.length === 0) {
      answer.push(n);
    } else {
      if (n === answer[answer.length - 1]) {
        answer.pop();
        answer.push(n);
      } else {
        answer.push(n);
      }
    }
  }
  return answer;
}
